src/lint/track_config.nim

```diff
@@ -393,13 +393,54 @@ proc checkExerciseConcepts(conceptExercises: seq[ConceptExercise];
                    "`slug` in the top-level `concepts` array"
         b.setFalseAndPrint(msg, path)
 
+proc checkForCycle(prerequisitesByConcept: Table[string, seq[string]];
+                   currentConcept: string;
+                   prereqPath: seq[string];
+                   conceptExerciseSlug: string;
+                   b, hadCycle: var bool; path: Path) =
+  ## Sets `b` to `false` if the given `conceptExerciseSlug` has a cycle due to
+  ## its `prerequisites`.
+  ##
+  ## An example of such a cycle:
+  ## - exercise slug A teaches concept A', and requires concept B'
+  ## - exercise slug B teaches concept B', and requires concept C'
+  ## - exercise slug C teaches concept C', and requires concept A'
+  ##
+  ## This is important to forbid because any exercise that is involved in such a
+  ## cycle would not be unlockable.
+  if hadCycle:
+    return
+
+  let updatedPrereqPath = prereqPath & @[currentConcept]
```

Comment:
Relatively costly allocations here - I think we can refactor later by making `prereqPath` a `var seq`.

Reply:
I think you need a copy for the recursion to work properly otherwise you'll keep appending to it (improperly). The other option is to remove things at various spots in the recursion. Either way, I think this can be deferred for later optimization.